src/db/nedb.js

```diff
@@ -57,13 +63,26 @@ class DBHelper {
     }
   }
 
-  static async insertOrUpdateCOracle(db, oracle) {
+  static async removeTopic(db, txid) {
     try {
+      const numRemoved = await db.remove({ txid }, {});
+      logger.debug(`Remove: ${numRemoved} Topic txid:${txid}`);
+    } catch (err) {
+      logger.error(`Remove: Topic txid:${txid}: ${err.message}`);
+    }
+  }
+
+  static async insertOrUpdateCOracle(db, oracle, queryTxid) {
+    try {
+      let txid = oracle.txid;
+      if (!_.isEmpty(queryTxid)) {
+        txid = queryTxid;
+      }
+
       await db.update(
-        { _id: oracle._id },
+        { txid },
         {
           $set: {
-            _id: oracle._id,
             txid: oracle.txid,
```

Comment:
`txid: txid`

Reply:
fixed

Reply:
same comment as above, we need to set the txid as the new createEvent txid